addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -116,8 +117,29 @@ def _search_valid(self, operator, value):
             or that don't need an allocation
             return [('id', domain_operator, [x['id'] for x in res])]
         """
-        date_to = self._context.get('default_date_from') or fields.Date.today().strftime('%Y-1-1')
-        date_from = self._context.get('default_date_to') or fields.Date.today().strftime('%Y-12-31')
+
+        if {'default_date_from', 'default_date_to', 'tz'} <= set(self._context):
+
+            default_date_from_str = self._context.get('default_date_to')  # sessions's local 07:00 given in UTC
+            default_date_to_str = self._context.get('default_date_from')  # sessions's local 19:00 given in UTC
+            context_timezone = pytz.timezone(self._context.get('tz'))     # user's tz
+
+            # Convert:  "2024-07-23 05:00:00"  ->  datetime.datetime(2024, 7, 23, 5, 0, tzinfo=<UTC>)
+            default_date_from_dt = datetime.datetime.fromisoformat(default_date_from_str).replace(tzinfo=pytz.UTC)
+            default_date_to_dt = datetime.datetime.fromisoformat(default_date_to_str).replace(tzinfo=pytz.UTC)
+
+            # Shift back to user's tz
+            default_date_from_dt_user_tz = default_date_from_dt.astimezone(context_timezone)
+            default_date_to_dt_user_tz = default_date_to_dt.astimezone(context_timezone)
+
+            # Cast: Datetime -> Date
+            date_from = default_date_from_dt_user_tz.strftime('%Y-%m-%d')
+            date_to = default_date_to_dt_user_tz.strftime('%Y-%m-%d')
```

Comment:
maybe this could use fields.Datetime/fields.Date method instead, but it is not much shorter, eg:

```suggestion
            # Convert:  "2024-07-23 05:00:00"  -> datetime.datetime(2024, 7, 23, 5, 0)
            default_date_from_dt = fields.Datetime.to_datetime(default_date_from_str)
            default_date_to_dt = fields.Datetime.to_datetime(default_date_to_str)

            # Cast: Datetime -> Date using user's tz
            date_from = fields.Date.context_today(self, default_date_from_dt)
            date_to = fields.Date.context_today(self, default_date_to_dt)
```

Reply:
Didn't know about them, that makes sense, Thanks :+1: